code/modules/antagonists/revolutionary/datum_revolutionary.dm

```diff
@@ -0,0 +1,58 @@
+/datum/antagonist/rev
+	name = "Revolutionary"
+	roundend_category = "revs"
+	job_rank = ROLE_REV
+	special_role = SPECIAL_ROLE_REV
+	give_objectives = FALSE
+	antag_hud_name = "hudrevolutionary"
+	antag_hud_type = ANTAG_HUD_REV
+	clown_gain_text = "Your newfound purpose has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself."
+	clown_removal_text = "As you feel your sense of purpose fade, you return to your own clumsy, clownish self."
+	wiki_page_name = "Revolution"
+	var/converted = TRUE
+
+/datum/antagonist/rev/on_gain()
+	create_team() // make sure theres a global rev team
+	owner.has_been_rev = TRUE
+	..()
+
+/datum/antagonist/rev/greet()
+	to_chat(owner.current, "<span class='userdanger'>You are now a revolutionary! Help your cause. \
+				Do not harm your fellow freedom fighters. You can identify your comrades by the red \"R\" icons, \
+				and your leaders by the blue \"R\" icons. Help them kill the heads to win the revolution!</span>")
+
+/datum/antagonist/rev/farewell()
+	if(owner && owner.current)
+		to_chat(owner.current,"<span class='userdanger'>You have been brainwashed! You are no longer a [special_role]! </span>")
+
+
+/datum/antagonist/rev/announce_objectives()
+
```

Comment:
```suggestion
/datum/antagonist/rev/announce_objectives()
	return
```

Reply:
I don't know why that was there, I think it was supposed to still announce objectives the normal way? I've just deleted it